src/components/QuestionTypeRangeChart.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component, Fragment } from 'react'
+import { interpolate } from 'd3-interpolate'
+import { css, merge } from 'glamor'
+
+import { Label, colors, mediaQueries } from '@project-r/styleguide'
+import { Chart } from '@project-r/styleguide/chart'
+
+import withT from '../lib/withT'
+
+const styles = {
+  chartLegend: css({
+    display: 'flex',
+    marginBottom: 20
+  }),
+  chartLegendColorSample: css({
+    width: 2,
+    marginRight: 4,
+    [mediaQueries.lUp]: {
+      width: 4
+    }
+  }),
+  chartLegendLabel: css({
+    paddingRight: 10
+  }),
+  rangeLegend: css({
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginTop: 10,
+    marginBottom: 20
+  })
+}
+
+class QuestionTypeRangeChart extends Component {
+  constructor(props) {
+    super(props)
+
+    this.colors = {
+      // Default colors
+      empty: 'white',
+      min: colors.secondaryBg,
+      max: colors.divider,
+      answer: colors.primary,
+
+      // Overwrite colors
+      ...props.colors
+    }
+  
+    this.getRangeColor = interpolate(this.colors.min, this.colors.max)
+  }
+
+  render () {
+    const { question, buckets, t } = this.props
+    if (!question || !question.rangeResults) {
+      return null
+    }
+
+    const { ticks, userAnswer, rangeResults } = question
+    const { histogram } = rangeResults
+
+    const value = userAnswer && userAnswer.payload && userAnswer.payload.value
+
+    const firstTick = ticks[0]
+    const lastTick = ticks[ticks.length - 1]
+
+    const legend = (buckets && buckets.filter(b => b.label).map(b => ({
+      label: b.label,
+      color: b.color
+    }))) || []
+
+    const values = histogram.map((bucket, index) => ({ time: index + 1, value: '1' }))
+
+    const max = histogram.reduce((prev, curr) => curr.count !== 0 && curr.count > prev.count ? curr : prev)
+    const min = histogram.reduce((prev, curr) => curr.count !== 0 && curr.count < prev.count ? curr : prev)
+
+    const colorRange = histogram.map((bucket, index) => {
+      const augmentedBucket = buckets && buckets.find(({ value }) =>
```

Comment:
value is already defined on line 60. This shadowing makes reading this quite hard.